api/base/middleware.py

```diff
@@ -47,3 +51,25 @@ def process_response(self, request, response):
                 raise err
         commands.disconnect()
         return response
+
+
+class FlaskRequestMiddleWare(object):
+    """
+    Push and pop new flask request contexts alongside Django requests
+
+    This is required to prevent caching issues with ModularODM, since (Flask)StoredObject is keyed on the flask
+    request object.
+    """
+    def process_request(self, request):
+        ## Called on every request, so self.flask_ctx should always be defined
+        self.flask_ctx = dummy_app.test_request_context()
+        self.flask_ctx.push()
+
+    def process_exception(self, request, exception):
```

Comment:
As I read the Django docs, `process_response` is called in all cases, while `process_exception` is only called in cases where an exception was raised.

Since we have a `process_response` defined, I don't believe it's necessary to define `process_exception`.

ref: https://docs.djangoproject.com/en/1.8/topics/http/middleware/#hooks-and-application-order